execution_engine/src/storage/global_state/lmdb.rs

```diff
@@ -252,7 +307,12 @@ mod tests {
         let protocol_data_store = Arc::new(
             LmdbProtocolDataStore::new(&environment, None, DatabaseFlags::empty()).unwrap(),
         );
-        let ret = LmdbGlobalState::empty(environment, trie_store, protocol_data_store).unwrap();
+        LmdbGlobalState::empty(environment, trie_store, protocol_data_store).unwrap()
+    }
```

Comment:
@afck This `new_empty_lmdb_global_state` makes an object where we can call `missing_descendant_trie_keys`, which I added validation support.

An approach to validating a state root hash against a lmdb file is to construct a `LmdbGlobalState` and run this method over it to make sure doesn't return an error or any missing nodes.